tools/default.nix

```diff
@@ -0,0 +1,46 @@
+{ pkgs ? import <nixpkgs> {
+    overlays = [
+      (import ../overlay.nix)
+    ];
+  }
+}:
+
+let
+  inherit (pkgs) lib;
+
+in
+{
+
+  flamegraph = let
+    runtimeDeps = lib.makeBinPath [
+      pkgs.flamegraph
+      pkgs.python3
+      pkgs.nix
+    ];
+
+    nixSrc = pkgs.runCommandNoCC "${pkgs.nix.name}-sources" {} ''
+      mkdir $out
+      tar -x --strip=1 -f ${pkgs.nix.src} -C $out
+    '';
+
+    srcPath = builtins.toString ../.;
+
+  in
+    pkgs.writeScriptBin "poetry2nix-flamegraph" ''
+      #!${pkgs.runtimeShell}
+      export PATH=${runtimeDeps}:$PATH
+
+      workdir=$(mktemp -d)
+      function cleanup {
+        rm -rf "$workdir"
```

Comment:
I tend to get nervous and `shellcheck` yells at me when I try to write `rm -rf $SOMETHING`. I know the scope is small here but it would be nice to rewrite this to something with a `/tmp` prefix: `rm -rf /tmp/$whatever`

Reply:
I don't know how else this would be implemented in a shell-script so I'll leave it for now. I'm also not a fan of the pattern and would much rather have something like python's `tempfile` which can clean things up when exiting the context manager.